sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -609,7 +629,8 @@ object ScalaJSPluginInternal {
         val libs =
           resolvedJSDependencies.value.data ++ scalaJSConfigurationLibs.value
         resolvedJSEnv.value match {
-          case env: LinkingUnitJSEnv =>
+          case env: LinkingUnitJSEnv
+              if scalaJSModuleKind.value == ModuleKind.NoModule =>
```

Comment:
That's actually precisely the issue. If I don't guard here, `env.loadLibs(libs).loadLinkingUnit(unit)` will have no idea that it is supposed to emit `unit` as a `ModuleKind`. So it will ignore it and desugar with `NoModule` anyway. The deeper problem is that the `LinkingUnitJSEnv` API was not designed to support `ModuleKind`s. I don't think it's worth evolving it to support them, though, since the only real use case is Rhino, and Rhino will disappear.